cmd/kubeadm/app/util/version.go

```diff
@@ -251,7 +251,7 @@ func validateStableVersion(remoteVersion, clientVersion string) (string, error)
 	if verClient.Major() < verRemote.Major() ||
 		(verClient.Major() == verRemote.Major()) && verClient.Minor() < verRemote.Minor() {
 		estimatedRelease := fmt.Sprintf("stable-%d.%d", verClient.Major(), verClient.Minor())
-		klog.Infof("remote version is much newer: %s; falling back to: %s", remoteVersion, estimatedRelease)
+		fmt.Printf("[warning] remote version is much newer: %s; falling back to: %s\n", remoteVersion, estimatedRelease)
```

Comment:
this is a problematic change, since it will go to stdout and the same logic is used by other commands not only upgrade.
klog prints to stderr.

Reply:
only upgrade use this

Reply:
kubernetesReleaseVersion calls validateStableVersion and kubernetesReleaseVersion is called in a number of commands:
mostly because of this which is relevant to all commands that accept --config:
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/cmd/kubeadm/app/util/config/common.go#L86

also as you can see we are already using klog to print similar warnings:
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/cmd/kubeadm/app/util/version.go#L102-L103

this is an issue that we logged klog vs printf, but we need a good design:
https://github.com/kubernetes/kubeadm/issues/1913

/close

Reply:
if you wish you can propose something for https://github.com/kubernetes/kubeadm/issues/1913 but we treat it as a low priority problem...

the mixed output is not great but we should also consider if we are breaking user expectations.